From 5a4cac73e1b5a96b495d4a3daff4eefb4120e8ba Mon Sep 17 00:00:00 2001
From: avignateintel <avignateintel@users.noreply.github.com>
Date: Mon, 1 Feb 2016 19:15:19 +0600
Subject: [PATCH] ART: Allow vendor to deploy optimized VM

This patch updates the main-level make file to consider using files
from a vendor defined VM directory. This is done so that vendor can
deploy optimized VM and compiler.

Change-Id: I9a8b9c108c176601f600ac8871300e56a514079a
---
 Android.mk | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/Android.mk b/Android.mk
index bb1334a..deed633 100644
--- a/Android.mk
+++ b/Android.mk
@@ -18,6 +18,15 @@ LOCAL_PATH := $(call my-dir)
 
 art_path := $(LOCAL_PATH)
 
+# Allow vendor to deploy another ART VM with proprietary optimizations.
+ifneq ($(VENDOR_ART_PATH),)
+  # Set this way to reduce conflicts in file.
+  art_path := $(VENDOR_ART_PATH)
+else
+  # If vendor is not deploying separate VM, just use ART.
+  VENDOR_ART_PATH := $(art_path)
+endif
+
 ########################################################################
 # clean-oat rules
 #
-- 
2.7.4

